offchain/danaswap-api/src/DanaSwap/Api.purs

```diff
@@ -33,6 +43,94 @@ type Protocol =
   , poolIdMP :: MintingPolicy
   }
 
+-- TODO should this be a newtype?
+type PoolId = TokenName
+
+getAllPools :: Protocol -> Contract () (Map TransactionInput TransactionOutputWithRefScript)
+getAllPools protocol@{ poolVal } =
```

Comment:
what does `poolVal` refer to? the `configUtxo` or `poolIdMP`?

Reply:
The pool address validator. Maybe it should be poolAdrVal?